sdk/python/kfp/components/component_decorator_test.py

```diff
@@ -0,0 +1,77 @@
+# Copyright 2022 The Kubeflow Authors
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import unittest
+
+from kfp.components import python_component
+from kfp.components.component_decorator import component
+
+
+def func(text: str) -> str:
+    """Function for lightweight component used throughout tests below."""
+    return text
+
+
+class TestComponentDecorator(unittest.TestCase):
+
+    def test_as_decorator_syntactic_sugar_no_args(self):
+
+        @component
+        def hello_world(text: str) -> str:
+            """Hello world component."""
+            return text
+
+        self.assertIsInstance(hello_world, python_component.PythonComponent)
+
+    def test_as_decorator_syntactic_sugar_some_args(self):
+
+        @component(base_image='python:3.9')
+        def hello_world(text: str) -> str:
+            """Hello world component."""
+            return text
+
+        self.assertIsInstance(hello_world, python_component.PythonComponent)
+
+    def test_no_args(self):
+
+        comp = component(func)
+        self.assertIsInstance(comp, python_component.PythonComponent)
+
+    def test_some_args(self):
+
+        comp = component(func, base_image='python:3.9')
+        self.assertIsInstance(comp, python_component.PythonComponent)
+
+    def test_packages_to_install(self):
+
+        comp = component(func, packages_to_install=['numpy', 'tensorflow'])
+        self.assertIsInstance(comp, python_component.PythonComponent)
+
+        concat_command = " ".join(
```

Comment:
nit: be consistent on using double quotes vs. single quotes. 

> Be consistent with your choice of string quote character within a file. Pick ' or " and stick with it. It is okay to use the other quote character on a string to avoid the need to backslash-escape quote characters within the string.

https://google.github.io/styleguide/pyguide.html#310-strings